.github/workflows/gradle.yml

```diff
@@ -0,0 +1,30 @@
+name: build-ci
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-18.04
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Java setup
+        uses: actions/setup-java@v1
+        with:
+          java-version: 8
+      - name: Build
+        run: chmod +x gradlew; ./gradlew setupCiWorkspace && ./gradlew build
+      - name: Upload build artifacts
+        uses: actions/upload-artifact@v1
+        with:
+          name: build-artifacts
+          path: build/libs
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
```

Comment:
Caching should be the first action, not the last action. The cache action restores from the cache and registers a "clean up" action that puts into the cache.

Reply:
Resolved